### DISCUSSION TITLE
Current isort config vs ruff

### DISCUSSION THREAD

**Discussion by martinhoyer** [teemtee/tmt#2003] [General]
With ruff being used now, I'd like to start a discussion about potentially dropping the isort package from pre-commit and rely on ruff's import sorting.

Advantages:
 - one less package in pre-commit (one less virualenv for individual devs and in CI runs)
 - ruff being faster

A few caveats though..
Currently, we cannot emulate the default isort config as it's being used in tmt:
1) different behaviour for imports wrap when they extend past the line length limit. See https://pycqa.github.io/isort/docs/configuration/multi_line_output_modes.html, isort uses `0 - Grid` by default, while ruff is equal to `3 - Vertical Hanging Indent`. Isort also doesn't do trailing commas by default, but that's beside the point.
In practice it looks like:
```
from tmt.utils import (Command, EnvironmentType, FmfContextType, Path, ShellScript,
                       WorkdirArgumentType, verdict)
```
vs
```
from tmt.utils import (
    Command,
    EnvironmentType,
    FmfContextType,
    Path,
    ShellScript,
    WorkdirArgumentType,
    verdict,
)
```
my subjective opinion is that Hanging is more readable, but I'm no authority on the matter.

2) Line length limit for isort is by default 79, which is how tmt imports are being sorted. Ruff has "global" line length limit.

If I leave default ruff settings and only add 'known-first-party' settings, so the `from tmt.` imports are sorted as they are now, the diff would looks like this:
https://github.com/teemtee/tmt/compare/main...martinhoyer:tmt:ruff_isort

What do you think?

**Comment by psss** [comment]
One import per line for long lists sounds perfectly fine. This will make patches adding one module to the import a bit more clear, as now rewrapping is a bit confusing.

**Comment by martinhoyer** [comment]
The isort->ruff move is being done in https://github.com/teemtee/tmt/pull/2014.
